storage/aptosdb/src/lib.rs

```diff
@@ -1761,11 +1770,22 @@ impl DbWriter for AptosDB {
                         &DbMetadataValue::Version(last_version),
                     )?;
 
-                    thread::scope(|s| {
-                        let t0 = s.spawn(|| self.state_kv_db.commit(last_version, state_kv_batch));
-                        let t1 = s.spawn(|| self.ledger_db.write_schemas(ledger_batch));
-                        t0.join().unwrap().unwrap();
-                        t1.join().unwrap().unwrap();
+                    COMMIT_POOL.scope_fifo(|s| {
+                        // TODO(grao): Consider propagating the error instead of panic, if necessary.
+                        s.spawn_fifo(|_| {
+                            let _timer = OTHER_TIMERS_SECONDS
+                                .with_label_values(&["save_transactions_commit___ledger_commit"])
+                                .start_timer();
+                            self.ledger_db.write_schemas(ledger_batch).unwrap();
```

Comment:
curious why we are using blocking thread vs async tasks sharing the thread here? seems write_schema is an I/O task suitable for async?

Reply:
We don't use async framework in storage so far, and RocksDB doesn't have async API, and we anyway rely on OS to do context switch, and ...

Reply:
not relevant to this review. But feel this can be generalized as util for db committing or even reading with multiple threads.